### PR TITLE
Rem 318

### PR DESCRIPTION
**Pull request by omarZrayka23**
After installing and testing agGridReact, everything functions properly on the client side. However, the server side obtains data directly from fetchgridata rather than via a user resource query.

### CODE REVIEW

**Comment by Ghinwa-Arab** [comment]
bugs in:
-checkbox (not binding from API checked data correctly + issues on checking + not submitting the correct chosen rows)
-delete not working (standard & dialog delete)

**Comment by NoraBitarSM** [comment]
1. open expense type screen, edit any record and open the info button, it will show an error (application error: a client side exception has occured). it works on deploy
image

2. 
image
can we check all and uncheck all (checkbox not rows) with this feature ? 

3. can we remove this feature for the columns whenever we like to ? or we're stick to it 
image

4. open currencies screen, it is showing all 186 currencies in one page even though the pagination is saying Displaying Records 1 - 50 Of 186

File: src/components/Shared/Table.js [hunk 1]
```diff
@@ -1,132 +1,146 @@
...
 
 const Table = ({
-  pagination = true,
-  paginationType = 'api',
-  height,
-  addedHeight = '0px',
-  actionColumnHeader = '',
-  showCheckboxColumn = false,
-  checkTitle = '',
+  fetchGridData,
```
Comment (hadichahine) [inline src/components/Shared/Table.js:33]: Remove

File: src/components/Shared/Table.js [hunk 2]
```diff
@@ -1,132 +1,146 @@
...
+  paginationType = '',
   viewCheckButtons = false,
-  ChangeCheckedRow,
+  showCheckboxColumn = false,
+  pagination = true,
+  handleCheckedRows,
```
Comment (hadichahine) [inline src/components/Shared/Table.js:38]: Remove

File: src/components/Shared/Table.js [hunk 3]
```diff
@@ -341,197 +364,148 @@ const Table = ({
...
+            field: 'checked',
+            cellRenderer: checkboxCellRenderer,
+            headerComponent: params => <Checkbox checked={checked} onChange={e => selectAll(params, e)} />,
+            suppressMenu: true // if i want to remove menu from header
+          }
+        ]
+      : []),
+    ...columns
+  ]
+  console.log('columns', columns)
```
Comment (hadichahine) [inline src/components/Shared/Table.js:466]: remove

File: src/components/Shared/Table.js [hunk 4]
```diff
@@ -1,132 +1,146 @@
...
-  const api = props?.api ? props?.api : props.paginationParameters
+  const pageSize = props?.pageSize || 100
+  const api = props?.api ? props?.api : props?.paginationParameters || ''
   const refetch = props?.refetch
+  const [gridData, setGridData] = useState({})
+  const [startAt, setStartAt] = useState(0)
+  const { languageId } = useContext(AuthContext)
+  const { platformLabels } = useContext(ControlContext)
   const maxAccess = props.maxAccess && props.maxAccess.record.maxAccess
   const columnsAccess = props.maxAccess && props.maxAccess.record.controls
```
Comment (hadichahine) [inline src/components/Shared/Table.js:50]: Make sure this isn't needed

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Pull request merged**